### DISCUSSION TITLE
Passing data from localStorage to getServerSideProps

### DISCUSSION THREAD

**Discussion by holycow1991** [vercel/next.js#16824] [Help] [answered]
Hello guys! 
I have kinda problem with my JWT token, today i tried to shoot an endpoint for my user data, and then i see 401!, what the heck? 
Okay, so i investigated, and there is a problem that my axios doesn't have Authorization token attached yet when i render this data.
Is there any way to smuggle a string around whole project, like localStorage, but it needs to be accessible on server side of application ? 
Cheeeers!!

**Comment by jamesmosier** [comment, marked as answer]
If you want to have data available on both the client and server you could use cookies. 

This [example](https://github.com/vercel/next.js/blob/canary/examples/with-passport/lib/auth-cookies.js) shows how cookies can be created/destroyed/read.

**Reply by holycow1991** [reply]
Thank's James, everything is clear now :)!

**Reply by holycow1991** [reply]
@jamesmosier  Another solution is to create some kind of Auth Context, which will store some auth data, and use it in getServerSideProps :)

**Reply by tjdam** [reply]
> example

I saw a similar piece of code is used in https://github.com/vercel/next.js/blob/canary/examples/api-routes-apollo-server-and-client-auth/ but I just realized used in the server side in the GraphQL resolvers... but what do I do if I want to have the cookies on server side of next but I'm using a GraphQL endpoint from an external source?

**Reply by alendra1945** [reply]
> @jamesmosier Another solution is to create some kind of Auth Context, which will store some auth data, and use it in getServerSideProps :)

can you give me example

**Reply by justincorrigible** [reply]
I'd also like seeing one, if you can spare the cycles. 🙏

**Reply by kwadoskii** [reply]
I ended up using cookies as instructed, thank you.
You have saved me after hours of debugging

**Reply by VenomFate-619** [reply]
to use that cookies it is necessary to set in getServerSideProps or I can set in client side and use it in server
